pkg/virt-handler/virt-chroot/virt-chroot.go

```diff
@@ -20,28 +20,45 @@
 package virt_chroot
 
 import (
+	"fmt"
+	"os"
 	"os/exec"
 	"strings"
 
+	ps "github.com/mitchellh/go-ps"
+
 	"kubevirt.io/kubevirt/pkg/safepath"
 	"kubevirt.io/kubevirt/pkg/unsafepath"
 )
 
 const (
 	binaryPath     = "/usr/bin/virt-chroot"
-	mountNamespace = "/proc/1/ns/mnt"
+	MountNamespace = "/proc/1/ns/mnt"
 )
 
+// GetHostNamespacePath - return the parent process
+// this is needed to handle the case kubernetes and kubevirt
+// running inside a container and the mount namespace is different
+// from the host/node.
+func GetHostNamespacePath() string {
+	pid := os.Getpid()
+	process, err := ps.FindProcess(pid)
+	if err != nil {
+		return MountNamespace
+	}
+	return fmt.Sprintf("/proc/%d/ns/mnt", process.PPid())
```

Comment:
You do not need to use the `ps` package here at all. Please refer to the comment https://github.com/kubevirt/kubevirt/pull/9668#discussion_r1266241467. This function can be just a one-liner:

```suggestion
	return fmt.Sprintf("/proc/%d/ns/mnt", os.Getppid())
```

Reply:
good point.